From 47bf546343f3390288d094e56dc7e8925a5c30f3 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 30 Mar 2026 03:24:59 +0000
Subject: [PATCH] links 2.30 fix build error.

---
 ftp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ftp.c b/ftp.c
index 7a8d05d..4b077d0 100644
--- a/ftp.c
+++ b/ftp.c
@@ -768,7 +768,7 @@ do {								\
 } while (0)
 		A(ftp_head);
 		ud = stracpy(get_url_data(c->url));
-		if (strchr(cast_const_char ud, POST_CHAR)) *strchr(cast_const_char ud, POST_CHAR) = 0;
+		if (strchr(cast_const_char ud, POST_CHAR)) *(unsigned char *)strchr(cast_const_char ud, POST_CHAR) = 0;
 		s0 = init_str();
 		s0l = 0;
 		add_conv_str(&s0, &s0l, ud, (int)strlen(cast_const_char ud), -1);
-- 
2.31.1

